client/source/js/modules/ui/type-selector/type-selector-ctrl.js

```diff
@@ -15,11 +16,32 @@ define(['./module', 'underscore'], function (module, _) {
       }
 
       $scope.$on('$stateChangeSuccess',function() {
-        showPartInViews('typeSelector', [
```

Comment:
When the reaction to an event is implemented on the spot of where the observation is made (like here is) what happens is that you make the code the opposite of intention revealing. 

When you care to name a method with a meaningful name (probably a the verb/action) what happens, due to the inverted pyramid principle, is that you make it more intention revealing, more friendly and fast to read and cheaper to maintain.

From line 19 to line 53 what action you intended for that controller?
That's the name of the method that should encapsulate that implementation (which, by the way, could have its proper comment which is not the case now)